x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/actions/SearchableSnapshotActionIT.java

```diff
@@ -503,6 +538,7 @@ public void testSearchableSnapshotActionOverridesMigrateAction() throws Exceptio
         indexDocument(client(), dataStream, true);
         String firstGenIndex = DataStream.getDefaultBackingIndexName(dataStream, 1L);
         Map<String, Object> indexSettings = getIndexSettingsAsMap(firstGenIndex);
+        // the searchable snapshot is allocated to the hot tier
```

Comment:
I think this is incorrect? This is asserting the managed index which hasn't yet been mounted as a searchable snapshot (we haven't rolled over yet). The assertion passes due to our default allocation of new indices to the hot tier. 

I'm not sure we need this IT anymore as the only time a mounted index will override the migrate action is in the frozen phase (the only place where we allow a partially mounted index and when we skip injecting the migrate action) - the allocation routing configured by the mounted index and the migrate action are identical so we can't distinguish who set it anymore. 

I suggest we convert this IT to a unit test for `MigrateAction#toSteps` in `MigrateActionTests` to make sure the migrate action is injected only when appropriate.

What do you think?

Reply:
Makes sense to me, @andreidan. What do you think of 46e4ba4?